### PR TITLE
[PKG-4489] lightgbm 4.3.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| main | [Anaconda Downloads](https://anaconda.org/main/lightgbm) | [Anaconda Version](https://anaconda.org/main/lightgbm) | [Anaconda Platforms](https://anaconda.org/main/lightgbm) | Anaconda License |

**Destination channel:**

### Links

- [PKG-4489](https://anaconda.atlassian.net/browse/PKG-4489)
- release-notes: https://github.com/Microsoft/LightGBM/releases
- release-notes-tagged: https://github.com/Microsoft/LightGBM/releases/tag/v4.3.0
- release-diff: https://github.com/Microsoft/LightGBM/compare/v4.1.0...v4.3.0
- license: https://github.com/microsoft/LightGBM/blob/master/LICENSE
- requirements-tagged:
  - https://github.com/microsoft/LightGBM/tree/v4.3.0/python-package
  - https://github.com/microsoft/LightGBM/blob/v4.3.0/CMakeLists.txt
  - https://github.com/microsoft/LightGBM/blob/v4.3.0/python-package/pyproject.toml

### Explanation of changes:

-

### Notes:

-

[PKG-4489]: https://anaconda.atlassian.net/browse/PKG-4489?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JamesRobertsonGames — APPROVED** [review]
Looks good to me! :+1:

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,22 +29,24 @@ requirements:
...
     # By default, mkl variants are picked. By picking the variant that uses openblas rather 
     # than mkl, intel-openmp will not be pulled in and so you will not have llvm-openmp / intel-openmp conflict. 
-    - blas=*=openblas # [osx and x86_64]
+    - blas=*=openblas  # [osx and x86_64]
     - scikit-build-core >=0.4.4
     - python
     - pip
-    - llvm-openmp  # [osx]
-    - libgomp      # [linux]
+    - llvm-openmp      # [osx]
```
Comment (cbouss) [inline recipe/meta.yaml:36]: llvm-openmp should be moved to build, has it has a strong run_exports.
See: https://github.com/AnacondaRecipes/openmp-feedstock/blob/master/recipe/meta.yaml#L32

Comment (cbouss) [reply recipe/meta.yaml:36]: `libgomp` can be removed and `_openmp_mutex # [linux]` added to run (or build, has it has a strong run_exports as well)..

Comment (skupr-anaconda) [reply recipe/meta.yaml:36]: Fixed

Comment (skupr-anaconda) [reply recipe/meta.yaml:36]: After removing `lbgomp`, ` osx-64` and `osx-arm64` failed:
`    ['  ERROR (lightgbm,lib/python3.9/site-packages/lightgbm/lib/lib_lightgbm.so): /private/var/folders/nz/j6p8yfhx1mv_0grj5xl4650h0000gp/T/abs_28g70qf1va/croot/lightgbm_1713967649472/_build_env/lib/libomp.dylib not found in packages, sysroot(s) nor the missing_dso_whitelist.\n.. is this binary repackaging?']`

But `libgomp` was only on linux.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,31 +20,34 @@ build:
 
 requirements:
   build:
-    - cmake >=3.8
-    - make                  # [unix]
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - cmake >=3.8
+    - make             # [unix]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: If you replace `make` with `ninja`, `scikit-build-core` should pick ninja and use it. It'll compile much faster than with make.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thanks! Done

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:36]: @cbouss I've tested it twice and it fails if `llvm-openmp` in `build` but in `host` is fine

Comment (skupr-anaconda) [reply recipe/meta.yaml:36]: I've added `_openmp_mutex # [linux]` to `run` and removed `libgomp`

Comment (cbouss) [reply recipe/meta.yaml:36]: I verified the correct run deps are set. It looks fine.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,49 +1,52 @@
...
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv -Ccmake.define.CMAKE_GENERATOR="Ninja" # [unix]
   missing_dso_whitelist:  # [s390x or osx]
     - "*/ld64.so.1"       # [s390x]
     - '*/libc++.1.dylib'  # [osx]
 
 requirements:
   build:
     - cmake >=3.8
-    - make                  # [unix]
+    - ninja     # [unix]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - ninja-base
```
You could use ninja for both win and unix.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

File: recipe/conda_build_config.yaml
Comment (cbouss) [inline recipe/conda_build_config.yaml:1]: This is still needed. The default cbc configuration is 2017, and you are using 2019 as cmake generator.

Comment (skupr-anaconda) [reply recipe/conda_build_config.yaml:1]: Fixed

**Review by cbouss — APPROVED** [review]

**Pull request merged**